build.sh

```diff
@@ -18,7 +18,7 @@ ARGS=$*
 # script, and that this script resides in the repo dir!
 REPODIR=$(cd $(dirname $0); pwd)
 
-VALIDARGS="clean libraft pyraft pylibraft docs tests bench clean -v -g --install --compile-libs --compile-nn --compile-dist --allgpuarch --sccache --no-nvtx --show_depr_warn -h --buildfaiss --minimal-deps"
+VALIDARGS="clean libraft pyraft pylibraft docs tests bench clean -v -g --install --compile-libs --compile-nn --compile-dist --allgpuarch --sccache --ccache --no-nvtx --show_depr_warn -h --buildfaiss --minimal-deps"
```

Comment:
In general I have recommended we switch over to a command line pattern like `--cache-tool=<program>` the reason for this is the following:

- It allows a user to specify either sccache or ccache.
- Since they provide a name or path we don't require the command to be on the PATH
- It removes the need to check if both options are enabled
- It unifies the `hasArg --ccache` and `hasArg --sccache` conditionals into a single branch
- It allows people to also use cache tools like `distcc`

Reply:
I like this approach! Thanks Robert. Let me update the PR.

Reply:
@robertmaynard can I get a re-review please?